### ISSUE TITLE
Complex if() freezes analysis

### ISSUE THREAD

**Issue by mpaluchowski** [cflint/CFLint#266]
Trying to analyze the following, rather complex, `if()` statement will freeze analysis. No error or anything, just hanging there until manually killed.

```
component {
  function foo() {
    if (keys[i] == sectionActionsName && !loadSectionFreeSkills && !loadSectionActions && !loadSectionActionsContact && !loadSectionActionsHistory && !loadSectionComment
        || keys[i] == sectionProfileName && !loadSectionProfile
        || keys[i] == sectionLanguagesName && !loadSectionLanguages
        || keys[i] == sectionExperienceWidgetName && !loadSectionProfile && !loadSectionExperience
        || keys[i] == sectionProfileName && !loadSectionProfile
        || keys[i] == sectionExperienceName && !loadSectionExperience
        || keys[i] == sectionSkillsName && !loadSectionSkills
        || keys[i] == sectionFreeSkillsName && !loadSectionFreeSkills
        || keys[i] == sectionFunctionsName && !loadSectionFunctions
        || keys[i] == sectionEducationName && !loadSectionEducation
        || keys[i] == sectionAttachmentName && !loadSectionAttachment
    ) {
        // do something
    }
  }
}
```

**Comment by jevans-gp** [comment]
that's terrible code, as a human i freeze reading it.

**Comment by mpaluchowski** [comment]
You're saying CFLint freezes in horror when it encounters that piece of code? A linter with feelings. How advanced! ;-)

Of course, I agree. And I'd love to see CFLint have a rule for that, ie. "if() condition has more than 5|configurable branches. Simplify it." It just shouldn't freeze or break otherwise.

**Comment by ryaneberly** [comment]
It's on the ANTLR parsing.   
Does this problem happen with v 1.0.1?  I wonder if the looser semi-colon handling is tying it in knots.

**Comment by mpaluchowski** [comment]
1.0.1 behaves the same, and so does 1.0.0. Both freeze.

**Comment by ryaneberly** [comment]
@mpaluchowski , Thanks for checking.  That helps - so it's not the semi-colon handling.  I'll try upgrading to antlr 4.7 before doing too much grammar debugging.

**Comment by ryaneberly** [comment]
This is fixed in cfparser 2.4.6  (coming to maven soon)
